packages/runtime-core/src/directives.ts

```diff
@@ -31,7 +31,7 @@ export type DirectiveHook<T = any> = (
   el: T,
   binding: DirectiveBinding,
   vnode: VNode<any, T>,
-  prevVNode: VNode | null
+  prevVNode: VNode<any, T> | null
```

Comment:
Technically only this is needed. This allows the user to author directives with specific el types. However all other functions are used internally only and adding generics doesn't offer any real benefits.

Reply:
done!